### PR TITLE
Error shouldn't cause Client Fail If Actor Non-Existent method called

### PR DESCRIPTION
**Pull request by DeepanshuA**
Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

# Description

Currently, for registered actors, if a non-existent method is called (let's say via postman), it makes App crash. This behavior is not correct and rather only an error with proper message needs to be sent.
This PR intends to fix this.

Got this response, after this change:

image

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #409 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Extended the documentation

### CODE REVIEW

**Comment by DeepanshuA** [comment]
> @DeepanshuA is this ready for review?

Ok, All tests seem to be passing on this PR.
I just went ahead to cross-check that e2e test that I have included has also passed.
But, it seems actor tests are not running as a part of all these tests.

I am checking my test separately now.

**Comment by XavierGeerinck** [comment]
Discussed: look at reverting the `package.json` back and see what the core issue is + wait on PR #381

File: src/implementation/Server/HTTPServer/actor.ts [hunk 1]
```diff
@@ -98,13 +99,18 @@ export default class HTTPServerActor implements IServerActor {
...
+    try {
+      const result = await ActorRuntime.getInstance(this.client.getDaprClient()).invoke(
+        actorTypeName,
+        actorId,
+        methodName,
+        dataSerialized,
+      );
+      return this.handleResult(res, result);
+    } catch (err) {
+      res.statusCode = HttpStatusCode.BAD_REQUEST;
```
Comment (shubham1172) [inline src/implementation/Server/HTTPServer/actor.ts:111]: Can `invoke` fail for any other thing than non-existent method scenario? Either we can make this a general error, like `HttpStatusCode.INTERNAL_SERVER_ERROR`, or we should only catch errors where a certain string is matched and then return `BAD_REQUEST` for it.

**Review by shubham1172 — APPROVED** [review]
Left a comment!

File: src/implementation/Server/HTTPServer/actor.ts (hunk 1 above)
Comment (DeepanshuA) [reply src/implementation/Server/HTTPServer/actor.ts]: Yup, it can fail for any other reason as well. I have changed it to INTERNAL_SERVER_ERROR.

**Review by shubham1172 — APPROVED** [review]
LGTM! We should also do this exercise for other actor methods like deactivate, but it can be done separately, fyi @XavierGeerinck

**Review by XavierGeerinck — APPROVED** [review]

**Comment by XavierGeerinck** [comment]
> 

Agreed

**Pull request merged**